File: android/versioned-abis/expoview-abi47_0_0/src/main/java/abi47_0_0/expo/modules/crypto/DigestOptions.kt
package abi47_0_0.expo.modules.crypto

import abi47_0_0.expo.modules.kotlin.records.Field
import abi47_0_0.expo.modules.kotlin.records.Record
import abi47_0_0.expo.modules.kotlin.types.Enumerable

class DigestOptions : Record {
  @Field
  var encoding: Encoding = Encoding.HEX

  enum class Encoding(val value: String) : Enumerable {
    HEX("hex"),
    BASE64("base64")
  }
}

File: packages/expo-crypto/ios/DigestOptions.swift
// Copyright 2022-present 650 Industries. All rights reserved.

import ExpoModulesCore

@Record
internal struct DigestOptions {
  var encoding: Encoding = .hex

  internal enum Encoding: String, Enumerable {
    case hex = "hex"
    case base64 = "base64"
  }
}
